pkg/netstore/netstore.go

```diff
@@ -47,17 +59,34 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 				if s.recoveryCallback == nil {
 					return nil, err
 				}
-				targets, err := sctx.GetTargets(ctx)
+				targets, err1 := sctx.GetTargets(ctx)
+				if err1 != nil {
+					return nil, err1
+				}
+				chunkC := make(chan swarm.Chunk, 1)
+				socAddress, err1 := s.recoveryCallback(ctx, addr, targets, chunkC)
+				if err1 != nil {
+					return nil, err1
+				}
+
+				// add the expected soc address and the channel in which the chunk is awaited
+				s.chunkChanMu.Lock()
+				s.chunkChanMap[socAddress.String()] = chunkC
+				s.chunkChanMu.Unlock()
+				defer delete(s.chunkChanMap, socAddress.String())
```

Comment:
This deferred delete on chunkChanMap is not protected by the lock.

Reply:
good catch. Thanks